Improvements to uRaft

uristdwarf opened the pull request:
This is a patch series. _**DO NOT MERGE OR SQUASH THIS, DO REBASE INSTEAD**_

Generally refactoring and fixing clang-tidy warnings/hints. This is in preparation for some fundamental changes to how uRaft works. It also improves logging (by logging to both syslog _and_ `stderr`) and fixes a bug with key-values not being correctly parsed from the uRaft helper script.

uristdwarf committed:
refactor(slogger): Add string representations of log levels

This for example allows using environment variables to set logging
levels.

uristdwarf committed:
refactor(uraft): General improvements to C++ code

* Fix clang-tidy linting warnings/hints
* Use slogger instead of only syslog
* Allow setting env variable to enable specific levels of logging

uristdwarf committed:
fix(uraft): Configuration file reading errors

The helper script couldn't correctly read the files unless there were no
spaces between the key values.

uristdwarf committed:
refactor(uraft): Improve the helper script

* Add more thorough logging (including logging to stderr)
* Allow passing some env variables through parent
* Allow setting log level by setting the SAUNAFS_LOG_LEVEL variable

Antuan-AE commented on src/slogger/slogger.cc:
```diff
@@ -44,6 +47,54 @@ static safs::log_level::LogLevel log_level_from_syslog(int priority) {
+std::string safs::log_level_to_string(safs::log_level::LogLevel level) {
+	using safs::log_level::LogLevel;
+	switch (level) {
+	case LogLevel::trace:
+		return "trace";
+	case LogLevel::debug:
+		return "debug";
+	case LogLevel::info:
+		return "info";
+	case LogLevel::warn:
+		return "warn";
+	case LogLevel::err:
+		return "err";
+	case LogLevel::critical:
+		return "crit";
+	case LogLevel::off:
+		break;
```
Consider adding `default` clause to avoid possible future errors in case an integer is cast to LogLevel type

uristdwarf replied:
I think in this case it should be an assert, as if it's anything, it's most likely a bug in the program

uristdwarf commented on src/uraft/uraftcontroller.cc:
```diff
@@ -1,41 +1,43 @@
+      check_node_status_timer_(ios),
+      cmd_timeout_timer_(ios),
+      command_pid_(-1),
+      command_type_(kCmdNone),
+      force_demote_(false),
+      node_alive_(false) {
+
+	opt_.check_node_status_period = DEFAULT_CHECK_NODE_STATUS_PERIOD;
+	opt_.check_cmd_status_period  = DEFAULT_CMD_STATUS_PERIOD;
+	opt_.getversion_timeout       = DEFAULT_GETVERSION_TIMEOUT;
```
This was done by clang format, not me

uristdwarf committed:
style(uraft): Convert to tab indendation from space

uristdwarf commented:
> @uristdwarf The github rebase button seems to do the opposite to what we want (it brings the changes in the opposite direction, leaving a weird shape in the graph), please, rebase from your computer if it is your will.

I actually want it to be fast-forwarded, so that all commits are preserved in dev. But maybe a merge will suffice.

uristdwarf committed:
style(uraft): Use colons instead of names

Co-authored-by: Walter Gomez <baldor@leil.io>

uristdwarf committed:
test: Fix NULL being used in std::string construction

uristdwarf committed:
fix(tests): Allow to change tests data path (#159)

When the binary for tests was moved to a different path it could not
find the tests bash script. This commit adds an environment variable
to allow the path change in run time.

Co-authored-by: Antuan <antuan@leil.io>
Co-authored-by: Crash <crash@leil.io>
